crates/wasi-http/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+use crate::component_impl::add_component_to_linker;
+pub use crate::r#struct::WasiHttp;
+
+wasmtime::component::bindgen!("proxy");
```

Comment:
If you set `async: true` then you can use `reqwest::Client` instead of `reqwest::blocking::Client`.

Reply:
Agreed - really, the only time we want to use the sync traits is to support hosts that arent running tokio. Since the reqwest blocking client is actually using tokio under the hood, using the sync trait means this library can't compose with the rest of the wasi libraries when run on tokio, but it also can't work for hosts that don't want to incur tokio, which is the worst of both worlds.

And once we make that change, I'd much prefer to drop down to just plain old hyper, because I don't think there's a ton of value in the reqwest async interface over the hyper primitives for this use case, and reqwest is responsible for at least half of the dependencies that need to be `cargo vet`ed in order to land this.

Reply:
I looked at hyper. It makes you do things like explicitly open a TCP connection before talking HTTP, it's ergonomics as a library are not great at first glance anyway.

I'm happy to do this as a later optimization, but I don't think that it impacts this PR, so I'd prefer to merge and iterate.

Reply:
In order to land the PR I have to audit all unvetted dependencies for supply chain safety. These audits are very time consuming - I have to review the source of the entire crate. We can only accept these audits if they are performed by core team members.  I can't make any kind of exception for this PR, or breeze through things because reqwest will be swapped out soon, because our audits are imported to be part of the supply chain safety of other projects.

Right now this PR introduces 49 unvetted dependencies into the project https://github.com/bytecodealliance/wasmtime/actions/runs/4430843524/jobs/7773090032?pr=5929. That represents at least a full day of very tedious work for me. Many of those dependencies are pulled in exclusively by the reqwest library, in order to provide functionality we don't need. (Because of the use by other projects, vetting considers all dependencies for all features and targets, even though our project doesn't need them.)

Therefore, we need to iterate on this before landing the PR. Like I said, I can send a PR to your branch to help with this transformation, because my time would be better spent on making the changes that need to happen anyway vs auditing packages we don't care about.

Reply:
The need to reduce the scope of rust vet makes total sense to me. I will look into switching this to Hyper.